NorthstarDedicatedTest/logrotation.cpp

```diff
@@ -0,0 +1,80 @@
+#include "pch.h"
+#include <filesystem>
+#include <string>
+#include <time.h>
+#include <iostream>
+#include <regex>
+#include "configurables.h"
+
+namespace fs = std::filesystem;
+using namespace std;
+
+const int seconds_in_day = 86400;
+
+double getLogAge(const fs::path path)
+{
+	time_t logtime;
+	time_t currenttime;
+	struct tm timeinfo = {0};
+	int year, month, day;
+	string filename = path.filename().string();
+	// Check if log has a correct name
+	if (regex_search(filename, regex("nslog([0-9]{2}-){2}[0-9]{4}\ [0-9]{2}(-[0-9]{2}){2}.txt")))
```

Comment:
The regular expressions should be compiled ahead of time, or at least only once.